agent/src/main/scala/za/co/absa/atum/agent/AtumContext.scala

```diff
@@ -0,0 +1,77 @@
+package za.co.absa.atum.agent
+
+import org.apache.spark.sql.DataFrame
+import za.co.absa.atum.agent.model.{MeasureResult, Measurement}
+
+/**
+ *  AtumContext: This class provides the methods to measure Spark `Dataframe`. Also allows to add/edit/remove measures.
+ *  @param measurements: A sequences of measurements.
+ */
+
+case class AtumContext(measurements: Set[Measurement] = Set()) {
+
+  def withMeasures(
+    measure: Measurement
+  ): AtumContext =
+    this.copy(measurements = Set(measure))
+
+  def withMeasures(
+    measures: Iterable[Measurement]
+  ): AtumContext =
+    this.copy(measurements = measures.toSet)
+
+  def withMeasuresAdded(
+    measure: Measurement
+  ): AtumContext =
+    this.copy(measurements = measurements + measure)
```

Comment:
so this is more like 'withMeasureAdded' ?

Reply:
yes but I think that when you are coding is more comfortable using the ide if the method is overloaded